src/core_modules/capture-core/components/WidgetReferral/ReferToOrgUnit/ReferToOrgUnit.container.js

```diff
@@ -0,0 +1,106 @@
+// @flow
+import React from 'react';
+import type { ComponentType } from 'react';
+import { withStyles } from '@material-ui/core';
+import classNames from 'classnames';
+import i18n from '@dhis2/d2-i18n';
+import { colors, spacers, spacersNum } from '@dhis2/ui';
+import { convertStringToDateFormat } from '../../../utils/converters/date';
+import { DateField, SingleOrgUnitSelectField } from '../../FormFields/New';
+import type { ReferralDataValueStates } from '../../WidgetEnrollmentEventNew/Validated/validated.types';
+
+const styles = {
+    wrapper: {
+        padding: `${spacers.dp16} 0`,
+        maxWidth: '55.75rem',
+    },
+    fieldWrapper: {
+        display: 'flex',
+        flexWrap: 'wrap',
+        alignItems: 'start',
+        justifyContent: 'space-between',
+        padding: `${spacers.dp8}  ${spacers.dp16}`,
+    },
+    fieldLabel: {
+        color: colors.grey900,
+        paddingTop: spacersNum.dp12,
+        paddingRight: spacersNum.dp16,
+        flexBasis: '200px',
+    },
+    fieldContent: {
+        flexBasis: '150px',
+        flexGrow: 1,
+    },
+    alternateColor: {
+        backgroundColor: colors.grey100,
+    },
+};
+
+type Props = {
+    referralDataValues: ReferralDataValueStates,
+    setReferralDataValues: (() => Object) => void,
+    ...CssClasses,
+}
+
+export const ReferToOrgUnitContainerPlain = ({ referralDataValues, setReferralDataValues, classes }: Props) => {
+    const onBlurDateField = (e) => {
+        setReferralDataValues(prevValues => ({
+            ...prevValues,
+            scheduledAt: convertStringToDateFormat(e),
+        }));
+    };
+
+    const onSelectOrgUnit = (e) => {
+        setReferralDataValues(prevValues => ({
+            ...prevValues,
+            orgUnit: {
+                id: e.id,
+                name: e.displayName,
+            },
+        }));
+    };
+
+    const onDeselectOrgUnit = () => {
+        setReferralDataValues(prevValues => ({
+            ...prevValues,
+            orgUnit: null,
+        }));
+    };
+
+    return (
+        <div className={classes.wrapper}>
+            <div className={classes.fieldWrapper}>
+                <div className={classes.fieldLabel}>
+                    {i18n.t('Schedule date / Due date', { interpolation: { escapeValue: false } })}
```

Comment:
Should use the description of the due date set in Maintenance. In other forms, it is used like this `formFoundation.getLabel('scheduledAt')`. Thanks!

Reply:
Yes, good point!
As we don't have a formFoundation for the constraining programStage, I built a hook to fetch it from the api. Could get it through IndexedDB, but it would require fetching a lot more data than we need. Open for suggestions.

Reply:
If the Widget is not self-contained, we tend to avoid extra fetches and use the metadata cached in IndexDB instead. IMHO even if more data is fetched once, this is preferable then fetching the scheduledAt metdata label each time `WidgetReferral` is mounted. WDYT?

Reply:
@eirikhaugstulen and I discussed this a bit earlier @simonadomnisoru. I agree, we prefer getting metadata from IndexedDB when we are outside of a self-contained Widget. 

At the same time I wasn't sure we should redo it now since this would be the way to do it if we make it self-contained (In self-contained mode we would probably be able to pass in the metadata, but if it is not passed in we would grab it from the api in the Widget itself).

But yeah, ideally we would grab it from IndexedDB like you said. We should also extend the IndexedDB interface with a callback method like we have on getAll to limit the data the calling method gets back (called `project` for getAll).

Even though we agreed earlier to leave it for now @eirikhaugstulen, maybe it's good to set precedence here and change it to IndexedDB after all (the `project` method you can decide if you want to implement here, or if we do it later)